cmd/object-api-utils.go

```diff
@@ -269,27 +268,3 @@ type byBucketName []BucketInfo
 func (d byBucketName) Len() int           { return len(d) }
 func (d byBucketName) Swap(i, j int)      { d[i], d[j] = d[j], d[i] }
 func (d byBucketName) Less(i, j int) bool { return d[i].Name < d[j].Name }
-
-// rangeReader returns a Reader that reads from r
-// but returns error after Max bytes read as errDataTooLarge.
-// but returns error if reader exits before reading Min bytes
-// errDataTooSmall.
-type rangeReader struct {
-	Reader io.Reader // underlying reader
-	Min    int64     // min bytes remaining
-	Max    int64     // max bytes remaining
-}
-
-func (l *rangeReader) Read(p []byte) (n int, err error) {
-	n, err = l.Reader.Read(p)
-	l.Max -= int64(n)
-	l.Min -= int64(n)
-	if l.Max < 0 {
-		// If more data is available than what is expected we return error.
-		return 0, errDataTooLarge
```

Comment:
Removal of this struct and implementation send this cleanup as PR separately, not as part of this PR.